c/compare.sh

```diff
@@ -0,0 +1,131 @@
+    echo "Invalid bit width" 1>&2
+    exit 1
+  fi
+
+  for ff in $(ls $(grep -v "^#" $f)) ; do
+    orig=$ff
+
+    # no original source available
+    if echo $ff | grep -q '^ddv-machzwd/' ; then
+      continue
+    elif [ $ff = "loops/s3_false-unreach-call.i" ] ; then
+      continue
+    elif echo $ff | egrep -q '^ldv-(linux-3.0|regression)/' ; then
+      # there is a related .cil.c file, but it doesn't necessarily match at all
+      # orig=$(echo $ff | sed 's/\.i$/.cil.c/')
+      continue
+    fi
+
+    # try to find a matching source file for a preprocessed task
+    if echo $ff | grep -q '\.c$' ; then
+      continue
+    elif echo $ff | grep -q '\.c\.i$' ; then
+      orig=$(echo $ff | sed 's/\.i$//')
+    elif echo $ff | grep -q '^ldv-memsafety/memleaks' ; then
+      orig=$(echo $ff | sed -e 's#/#/memleaks-notpreprocessed/#' -e 's/\.i$/.c/')
+    else
+      orig=$(echo $ff | sed 's/\.i$/.c/')
+    fi
+
+    if [ ! -s $orig ] ; then
+      echo "No original source of $ff found" 1>&2
+      exit 1
+    fi
+
+    blacklist="\
+      array-memsafety/cstrlen_unsafe_false-valid-deref.i \
+      array-examples/standard_copyInitSum_true-unreach-call_ground.i \
+      array-examples/standard_seq_init_true-unreach-call_ground.i \
+      bitvector/byte_add_1_true-unreach-call.i \
+      bitvector/byte_add_false-unreach-call.i \
+      bitvector/gcd_1_true-unreach-call.i \
+      bitvector/gcd_2_true-unreach-call.i \
+      bitvector/gcd_3_true-unreach-call.i \
+      bitvector/interleave_bits_true-unreach-call.i \
+      bitvector-loops/verisec_sendmail__tTflag_arr_one_loop_false-unreach-call.i \
+      bitvector/modulus_true-unreach-call.i \
+      bitvector/num_conversion_2_true-unreach-call.i \
+      floats-cbmc-regression/float-flags-simp1_true-unreach-call.i \
+      floats-cbmc-regression/float-no-simp4_true-unreach-call.i \
+      floats-esbmc-regression/trunc_nondet_2_true-unreach-call.i \
+      seq-pthread/cs_dekker_true-unreach-call.i \
+      seq-pthread/cs_fib_longer_true-unreach-call.i \
+      seq-pthread/cs_fib_true-unreach-call.i \
+      seq-pthread/cs_lamport_true-unreach-call.i \
+      seq-pthread/cs_peterson_true-unreach-call.i \
+      seq-pthread/cs_queue_true-unreach-call.i \
+      seq-pthread/cs_read_write_lock_true-unreach-call.i \
+      seq-pthread/cs_stack_true-unreach-call.i \
+      seq-pthread/cs_stateful_true-unreach-call.i \
+      seq-pthread/cs_sync_true-unreach-call.i \
+      seq-pthread/cs_szymanski_true-unreach-call.i \
+      seq-pthread/cs_time_var_mutex_true-unreach-call.i \
+      "
+
+    blacklisted=0
+    for b in $blacklist ; do
+      if [ $b = $ff ] ; then
+        blacklisted=1
+        break
+      fi
+    done
+
+    [ $blacklisted -eq 0 ] || continue
+
+    goto-cc -m$bits $orig
+    goto-cc -m$bits $ff -o b.out
+    c=$(goto-diff --verbosity 2 -u a.out b.out | wc -l)
+    if [ $c -ne 0 ] ; then
+      goto-diff -u a.out b.out
+      rm -f a.out b.out
+      echo $ff
+      exit 1
```

Comment:
Aborting early is ok. But maybe the user wants a full list of invalid files (see `check.py`). Does someone want to change this? Otherwise just keep as is.